### PR TITLE
Refactor UB04 837i X12 feature to make work again.

### LINKED ISSUE #7413: UB04 837i X12 feature
### Describe the bug
Feature is broken in every way.
- Fee Sheet doesn't save revenue codes
- Custom Codes doesn't save or select revenue codes.
- UB04 form doesn't render
- 837i X12 claim doesn't generate due to missing revenue code plus.

### To Reproduce
Turn on UB04 and 837i settings in Config
Create a claim
Use Institutional options from Billing Manager

### Client configuration

- Browser:
- OpenEMR version:
- Operating system:

### PR DESCRIPTION
**Pull request by sjpadgett**
Fixes #7413 

#### Short description of what this resolves:
Many reverts and fixes to project.

#### Changes proposed in this pull request:
Fixed from issues
- Fee Sheet doesn't save revenue codes
- Custom Codes doesn't save or select revenue codes.
- UB04 form doesn't render
- 837i X12 claim doesn't generate due to missing revenue code plus.

### CODE REVIEW

File: src/Billing/BillingProcessor/BillingClaimBatch.php [hunk 1]
```diff
@@ -20,8 +20,6 @@
 
 namespace OpenEMR\Billing\BillingProcessor;
 
-use OpenEMR\Billing\BillingProcessor\BillingClaimBatchControlNumber;
```
Comment (stephenwaite) [inline src/Billing/BillingProcessor/BillingClaimBatch.php:23]: hi @sjpadgett , great work getting this going again

pretty sure need this when using it below in lines 61 and 64

Comment (sjpadgett) [reply src/Billing/BillingProcessor/BillingClaimBatch.php:23]: nope it's already in scope

Comment (stephenwaite) [reply src/Billing/BillingProcessor/BillingClaimBatch.php:23]: oh, nvm, since it's in the same namespace :)

Comment (sjpadgett) [reply src/Billing/BillingProcessor/BillingClaimBatch.php:23]: Also a FYI the latest ci will optimize uses and removed unused automatically so be sure to keep an eye on and verify removals yourself.

Comment (stephenwaite) [reply src/Billing/BillingProcessor/BillingClaimBatch.php:23]: okay for phpstorm?

File: interface/billing/ub04_dispose.php [hunk 2]
```diff
@@ -347,10 +350,10 @@ function get_ub04_array($pid, $encounter, &$log = "")
...
     }
     // @todo Not sure
-    $ub04id[367] = $diagnosis[0] ? substr($diagnosis[0], 0, 7) : ''; /* 69. ADMITTING DIAGNOSIS CODE */
-    $ub04id[368] = $diagnosis[1] ? substr($diagnosis[1], 0, 7) : ''; /* 70a. PATIENT'S REASON FOR VISIT */
-    $ub04id[369] = $diagnosis[2] ? substr($diagnosis[2], 0, 7) : ''; /* 70b. PATIENT'S REASON FOR VISIT */
-    $ub04id[370] = $diagnosis[3] ? substr($diagnosis[3], 0, 7) : ''; /* 70c. PATIENT'S REASON FOR VISIT */
+    $ub04id[367] = $diagnosis[0] ?? '' ? substr($diagnosis[0], 0, 7) : ''; /* 69. ADMITTING DIAGNOSIS CODE */
+    $ub04id[368] = $diagnosis[1] ?? '' ? substr($diagnosis[1], 0, 7) : ''; /* 70a. PATIENT'S REASON FOR VISIT */
+    $ub04id[369] = $diagnosis[2] ?? '' ? substr($diagnosis[2], 0, 7) : ''; /* 70b. PATIENT'S REASON FOR VISIT */
+    $ub04id[370] = $diagnosis[3] ?? '' ? substr($diagnosis[3], 0, 7) : ''; /* 70c. PATIENT'S REASON FOR VISIT */
```
Comment (stephenwaite) [inline interface/billing/ub04_dispose.php:356]: seal-dunk

**Review by stephenwaite — APPROVED** [review]

**Comment by sjpadgett** [comment]
image

image

I'm merging so I can get it out to customer to test the 837i claim X12. Hopefully I won't need to come back to this in another 7 years!:)

**Pull request merged**